### PR TITLE
raftstore-v2: only send clean snapshot

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?
Issue Number: Ref #12842 

What's Changed:

```commit-message
When the tablet contains dirty data right after split, generating snapshot may
just a waste. On the other hand, split usually happens on all peers, so delay
it a bit actually makes all peers more likely to be initialized by split. So
this PR rejects generating snapshot when it detects it still has dirty data.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore-v2/src/operation/ready/mod.rs [hunk 1]
```diff
@@ -80,7 +81,25 @@ impl<'a, EK: KvEngine, ER: RaftEngine, T: Transport> PeerFsmDelegate<'a, EK, ER,
 }
 
 impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
-    pub fn maybe_pause_for_recovery(&mut self) -> bool {
+    pub fn maybe_pause_for_recovery<T>(&mut self, store_ctx: &mut StoreContext<EK, ER, T>) -> bool {
+        // The task needs to be scheduled even if the tablet may be replaced during
+        // recovery. Otherwise if there are merges during recovery, the FSM may
+        // be paused forever.
```
Comment (tabokie) [inline components/raftstore-v2/src/operation/ready/mod.rs:87]: Don't understand this last line of comment.

**Review by tabokie — APPROVED** [review]

File: components/raftstore-v2/src/operation/ready/mod.rs (hunk 1 above)
Comment (BusyJay) [reply components/raftstore-v2/src/operation/ready/mod.rs:87]: Because merge needs to wait till two tablets become clean before execution.

**Review by ethercflow — APPROVED** [review]
LGTM

**Pull request merged**